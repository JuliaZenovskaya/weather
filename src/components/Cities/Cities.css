.city_input {
  float: left;
  margin-top: 30px;
}

.fav_cities {
  float: left;
  width: 70%;
  font-size: 20px;
  font-weight: bold;
  margin-top: 30px;
}
