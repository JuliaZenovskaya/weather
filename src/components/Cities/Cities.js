import React from "react";
import { connect } from "react-redux";
import Weather from '../Weather/Weather';
import { addCity, deleteCity, getWeatherByCity } from '../../actions/сitiesAction';
import './Cities.css';

class Cities extends React.Component {
  render() {
    return (
      <div>
      <div>
        <div className="fav_cities">Избранное</div>
        <form onSubmit={(e) => this.addNewCity(e)}>
          <input className="city_input" type="text" name="city" required />
          <input className="city_button" type="submit" value="Добавить"/>
        </form>
      </div >
        {this.props.error && <div className="error">Error: {this.props.error}</div>}
        <div  className="weather">
          {
            [...this.props.cities.entries()].map((entry) => {
              return (
                <Weather
                  key={entry[0]}
                  getWeather={() => this.props.getWeatherByCity(entry[0])}
                  onDelete={() => this.props.deleteCity(entry[0])}
                  weather={entry[1]}/>
              );
            })
          }
        </div>
      </div>
    );
  }

  addNewCity(e) {
    e.preventDefault();
    this.props.addCity(e.currentTarget.elements.city.value);
  }
}


function mapStateToProps(state) {
  return {
    cities: state.fav_cities.cities,
    error: state.fav_cities.error
  };
}

function mapDispatchToProps(dispatch) {
  return {
    addCity: (city) => {
      dispatch(addCity(city));
    },

    deleteCity: (city) => {
      dispatch(deleteCity(city));
    },

    getWeatherByCity: (city) => {
      dispatch(getWeatherByCity(city));
    }
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Cities);
