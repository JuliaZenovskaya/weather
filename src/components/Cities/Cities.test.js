import React from "react";
import Cities from "./Cities";
import renderer from "react-test-renderer";
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux'
import thunk from "redux-thunk";

const tempStore = configureMockStore([thunk]);
const response = {
    "coord":{"lon":37.62,"lat":55.75},
    "weather":[{"id":701,"main":"Mist","description":"mist","icon":"50n"}],
    "base":"stations",
    "main":{"temp":4.38,"pressure":1015,"humidity":100,"temp_min":2,"temp_max":5.56},
    "visibility":3000,
    "wind":{"speed":4,"deg":90},
    "clouds":{"all":90},
    "dt":1573170827,
    "sys":{"type":1,"id":9027,"country":"RU","sunrise":1573188558,"sunset":1573220232},
    "timezone":10800,
    "id":524901,
    "name":"Moscow",
    "cod":200
  };

describe("Correct display for cities", () => {
  it("There aren't any cities in the store", () => {
    let cities = new Map();
      const store = tempStore({
        fav_cities: {
          cities: cities
          }
        });

      const htmlTree = renderer.create(
        <Provider store={store}>
          <Cities/>
       </Provider>).toJSON();
      expect(htmlTree).toMatchSnapshot();
  });

  it("There are some cities in the store", () => {
    let cities = new Map([["Moscow", response]]);
      const store = tempStore({
        fav_cities: {
              cities: cities
          }
        });

      const htmlTree = renderer.create(
        <Provider store={store}>
          <Cities/>
       </Provider>).toJSON();
      expect(htmlTree).toMatchSnapshot();
  });
});

describe("Some error with displaying cities", () => {
  it("There aren't any cities in the store", () => {
    let cities = new Map();
      const store = tempStore({
        fav_cities: {
          cities: cities,
          error: 'error'
          }
        });

      const htmlTree = renderer.create(
        <Provider store={store}>
          <Cities/>
       </Provider>).toJSON();
      expect(htmlTree).toMatchSnapshot();
  });

  it("There are some cities in the store", () => {
    let cities = new Map([["Moscow", response], ["AnotherMoscow", response]]);
      const store = tempStore({
        fav_cities: {
          cities: cities,
          error: 'error'
          }
        });

      const htmlTree = renderer.create(
        <Provider store={store}>
          <Cities/>
       </Provider>).toJSON();
      expect(htmlTree).toMatchSnapshot();
  });})
