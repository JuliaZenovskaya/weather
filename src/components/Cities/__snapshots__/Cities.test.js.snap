// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct display for cities There are some cities in the store 1`] = `
<div>
  <div>
    <div
      className="fav_cities"
    >
      Избранное
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="city_input"
        name="city"
        required={true}
        type="text"
      />
      <input
        className="city_button"
        type="submit"
        value="Добавить"
      />
    </form>
  </div>
  <div
    className="weather"
  >
    <div
      className="weather"
    >
      <div
        className="left"
      >
        <div>
          Moscow
        </div>
        <img
          src="https://openweathermap.org/img/wn/50n.png"
        />
        <div>
          -269
           °C
        </div>
        <button
          className="button"
          onClick={[Function]}
        >
          Удалить
        </button>
      </div>
      <div
        className="right"
      >
        <div
          className="pressure"
        >
          Давление: 
          1015
           hPa
        </div>
        <div
          className="humidity"
        >
          Влажность: 
          100
          %
        </div>
        <div
          className="clouds"
        >
          Облачность: 
          mist
        </div>
        <div
          className="wind"
        >
          Ветер: 
          4
           м/с
        </div>
        <div
          className="coords"
        >
          Координаты: [
          37.62
          , 
          55.75
          ]
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Correct display for cities There aren't any cities in the store 1`] = `
<div>
  <div>
    <div
      className="fav_cities"
    >
      Избранное
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="city_input"
        name="city"
        required={true}
        type="text"
      />
      <input
        className="city_button"
        type="submit"
        value="Добавить"
      />
    </form>
  </div>
  <div
    className="weather"
  />
</div>
`;

exports[`Some error with displaying cities There are some cities in the store 1`] = `
<div>
  <div>
    <div
      className="fav_cities"
    >
      Избранное
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="city_input"
        name="city"
        required={true}
        type="text"
      />
      <input
        className="city_button"
        type="submit"
        value="Добавить"
      />
    </form>
  </div>
  <div
    className="error"
  >
    Error: 
    error
  </div>
  <div
    className="weather"
  >
    <div
      className="weather"
    >
      <div
        className="left"
      >
        <div>
          Moscow
        </div>
        <img
          src="https://openweathermap.org/img/wn/50n.png"
        />
        <div>
          -269
           °C
        </div>
        <button
          className="button"
          onClick={[Function]}
        >
          Удалить
        </button>
      </div>
      <div
        className="right"
      >
        <div
          className="pressure"
        >
          Давление: 
          1015
           hPa
        </div>
        <div
          className="humidity"
        >
          Влажность: 
          100
          %
        </div>
        <div
          className="clouds"
        >
          Облачность: 
          mist
        </div>
        <div
          className="wind"
        >
          Ветер: 
          4
           м/с
        </div>
        <div
          className="coords"
        >
          Координаты: [
          37.62
          , 
          55.75
          ]
        </div>
      </div>
    </div>
    <div
      className="weather"
    >
      <div
        className="left"
      >
        <div>
          Moscow
        </div>
        <img
          src="https://openweathermap.org/img/wn/50n.png"
        />
        <div>
          -269
           °C
        </div>
        <button
          className="button"
          onClick={[Function]}
        >
          Удалить
        </button>
      </div>
      <div
        className="right"
      >
        <div
          className="pressure"
        >
          Давление: 
          1015
           hPa
        </div>
        <div
          className="humidity"
        >
          Влажность: 
          100
          %
        </div>
        <div
          className="clouds"
        >
          Облачность: 
          mist
        </div>
        <div
          className="wind"
        >
          Ветер: 
          4
           м/с
        </div>
        <div
          className="coords"
        >
          Координаты: [
          37.62
          , 
          55.75
          ]
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Some error with displaying cities There aren't any cities in the store 1`] = `
<div>
  <div>
    <div
      className="fav_cities"
    >
      Избранное
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="city_input"
        name="city"
        required={true}
        type="text"
      />
      <input
        className="city_button"
        type="submit"
        value="Добавить"
      />
    </form>
  </div>
  <div
    className="error"
  >
    Error: 
    error
  </div>
  <div
    className="weather"
  />
</div>
`;
